orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/image/UpsertMachineImageTagsTask.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.image;
+
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.orca.DefaultTaskResult;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.Task;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.clouddriver.KatoService;
+import com.netflix.spinnaker.orca.clouddriver.model.TaskId;
+import com.netflix.spinnaker.orca.clouddriver.tasks.AbstractCloudProviderAwareTask;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class UpsertMachineImageTagsTask extends AbstractCloudProviderAwareTask implements Task {
+  @Autowired
+  KatoService kato;
+
+  @Autowired
+  List<MachineImageTagger> machineImageTaggers;
+
+  @Override
+  public TaskResult execute(Stage stage) {
+    String cloudProvider = getCloudProvider(stage);
+
+    MachineImageTagger tagger = machineImageTaggers.stream()
```

Comment:
feels like it might be useful to have a list of the previous tag values written to the context